src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackageCommandRunner.cs

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using NuGet.CommandLine.XPlat.Utility;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+/// <summary>
+/// A struct to simplify holding all of the information
+/// about a package reference when using list
+/// </summary>
+public struct PRPackage
+{
+    public string package;
+    public VersionRange requestedVer;
+    public string requestedVerStr;
+    public NuGetVersion resolvedVer;
+    public NuGetVersion suggestedVer;
+    public bool deprecated;
+    public bool autoRef;
+}
+
+namespace NuGet.CommandLine.XPlat
+{
+    public class ListPackageCommandRunner : IListPackageCommandRunner
+    {
+        private bool _autoReferenceFound = false;
+
+        public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs, MSBuildAPIUtility msBuild)
```

Comment:
why are you passing a utility class here? you don't need this

Reply:
Some of my functions in the utility class are not static. Should I change them to static and get rid of creating an instance of the class?